File: c++/21921.cpp
#include <iostream>

using namespace std;

int n, x;
int arr[250001];

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  // freopen("example.txt", "r", stdin);

  cin >> n >> x;
  for (int i = 0; i < n; i++) {
    cin >> arr[i];
  }

  int visitedNum = 0;
  for (int i = 0; i < x; i++) {
    visitedNum += arr[i];
  }
  int maxVisitedNum = visitedNum;
  int cnt = 1;

  for (int i = 0; i < n - x; i++) {
    visitedNum -= arr[i];
    visitedNum += arr[i + x];
    if (visitedNum > maxVisitedNum) {
      maxVisitedNum = visitedNum;
      cnt = 1;
    } else if (visitedNum == maxVisitedNum)
      cnt++;
  }

  if (maxVisitedNum == 0)
    cout << "SAD";
  else
    cout << maxVisitedNum << '\n' << cnt;
  cout << '\n';

  return 0;
}

File: js/21921.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, x] = inputs[0].split(' ').map(Number);
const arr = inputs[1].split(' ').map(Number);

let visitedNum = 0;
for (let i = 0; i < x; i++) {
  visitedNum += arr[i];
}
let maxVisitedNum = visitedNum;
let cnt = 1;

for (let i = 0; i < n - x; i++) {
  visitedNum -= arr[i];
  visitedNum += arr[i + x];
  if (maxVisitedNum < visitedNum) {
    maxVisitedNum = visitedNum;
    cnt = 1;
  } else if (maxVisitedNum === visitedNum) cnt++;
}

if (maxVisitedNum === 0) console.log('SAD');
else {
  console.log(`${maxVisitedNum}\n${cnt}`);
}
